.tours/particle-indexing.tour

```diff
@@ -0,0 +1,76 @@
+      "line": 208
+    },
+    {
+      "file": "yt/geometry/particle_geometry_handler.py",
+      "description": "This line, which happens *outside* the loop over particle positions, compresses the bitarrays.  That way we keep our memory usage to a minimum.",
+      "line": 209
+    },
+    {
+      "file": "yt/geometry/particle_geometry_handler.py",
+      "description": "This line is crucial for the next step of computing the refined indexes.  It conducts a set of logical operations to see which bits in the array are set *more than once* -- which means that there's more than one place you'll have to look if you're looking for particles in that region.  It's in those places that we make a second, more refined index.",
+      "line": 211
+    },
+    {
+      "file": "yt/geometry/particle_geometry_handler.py",
+      "description": "This is the tricky part, where we conduct a second bitmapping operation.\n\nHere, we look at those places where collisions in the coarse index have occurred.  We want to do our best to disambiguate the regions that different data files encompass, so in all of those regions, we insert a new *subdivided* region.  So for instance, if the region in `i,j,k` of the coarse index is touched by a couple files, we insert in `i,j,k` a whole new index, this time based on `index_order2`.  And, we index that.  Because we're using compressed indexes, this usually does not take up that much memory, but it can get unwieldy in those cases where you have particles with big smoothing lengths that need to be taken into account.\n\nThis loop requires a bit more of a dance, because for each collision in the coarse index, we have to construct a refined index.  So the memory usage is a bit higher here, because we can't compress the arrays until we're totally done with them.\n\nBut, at the end of this, what we have is a set of bitmaps -- one for each data file -- that tell us where the data file exerts its influence.  And then, wherever more than one data file exerts its influence, we have *another* set of bitmaps for each of *those* data files that tell us which sub-regions are impacted.\n\nThis then gets used whenever we want to read data, to tell us which data files we have to look at.  For sub-selecting from a really big set of particles, we can save an awful lot of time and IO!",
```

Comment:
This sentence "This then gets used whenever we want to read data, to tell us which data files we have to look at." could use a very brief hint of how it's used. Something like:

"This then gets used whenever we want to read data, to tell us which data files we have to look at. The data files that are flagged make up the `chunk` iterator used by the IO handler to do the actual reading of data, a subject for a future tour."